R/expect-setequal.R

```diff
@@ -32,41 +32,37 @@ expect_setequal <- function(object, expected) {
   }
 
   act_miss <- !act$val %in% exp$val
-  if (any(act_miss)) {
-    fail(
-      paste0(act$lab, "[", locations(act_miss), "] absent from ", exp$lab)
-    )
-  }
-
   exp_miss <- !exp$val %in% act$val
-  if (any(exp_miss)) {
-    fail(
-      paste0(exp$lab, "[", locations(exp_miss), "] absent from ", act$lab)
-    )
-  }
 
-  if (!any(exp_miss) && !any(act_miss)) {
+  if (any(exp_miss) || any(act_miss)) {
+    fail(paste0(
+      act$lab, " and ", exp$lab, " don't have the same values\n",
+      if (any(exp_miss))
+        paste0("* Missing from expected: ",  strings(act$val[act_miss]), "\n"),
+      if (any(act_miss))
+        paste0("* Missing from actual: ",  strings(exp$val[exp_miss]), "\n")
```

Comment:
How would you feel about framing this as `Only in expected:` instead of `Missing from expected:`?

I have trouble reading this example:

```r
x <- c("a", "b")
y <- c("b", "c")

expect_snapshot_failure(expect_setequal(x, y))
#> `x` and `y` don't have the same values
#> * Missing from expected: "a"
#> * Missing from actual: "c"
```

and I think it is because my brain is juggling too many things. I have to realize that `y` is "expected" but then look over at `x` to see that it did indeed have `"a"` which isn't in `y`.

This makes more sense to me

```r
x <- c("a", "b")
y <- c("b", "c")

expect_snapshot_failure(expect_setequal(x, y))
#> `x` (`actual`) and `y` (`expected`) don't have the same values
#> * Only in `actual`: "a"
#> * Only in `expected`: "c"
```

I also like seeing ``` `x` (`actual`) ``` because I can't ever remember which is which. It is also what `expect_equal()` and `expect_identical()` seem to do

```r
> testthat::expect_identical(1, 1.5)
Error: 1 (`actual`) not identical to 1.5 (`expected`).

  `actual`: 1.0
`expected`: 1.5
> testthat::expect_equal(1, 1.5)
Error: 1 (`actual`) not equal to 1.5 (`expected`).

  `actual`: 1.0
`expected`: 1.5
```

Reply:
I like it.